packages/govern-core/contracts/Govern.sol

```diff
@@ -5,28 +5,36 @@
 pragma solidity 0.6.8;
 pragma experimental ABIEncoderV2;
 
-import "erc3k/contracts/IERC3000.sol";
+import "erc3k/contracts/ERC3000.sol";
 import "erc3k/contracts/ERC3000Executor.sol";
 
 import "@aragon/govern-contract-utils/contracts/acl/ACL.sol";
+import "@aragon/govern-contract-utils/contracts/adaptative-erc165/AdaptativeERC165.sol";
 import "@aragon/govern-contract-utils/contracts/bitmaps/BitmapLib.sol";
 
-contract Govern is ERC3000Executor, ACL {
+contract Govern is AdaptativeERC165, ERC3000Executor, ACL {
     using BitmapLib for bytes32;
 
     bytes4 internal constant EXEC_ROLE = this.exec.selector;
+    bytes4 internal constant REGISTER_ROLE = this.registerStandardAndCallback.selector;
     uint256 internal constant MAX_ACTIONS = 256;
 
     event ETHDeposited(address indexed sender, uint256 value);
 
-    constructor(IERC3000 _initialExecutor) ACL(address(this)) public {
+    constructor(ERC3000 _initialExecutor) ACL(address(this)) public {
         _grant(EXEC_ROLE, address(_initialExecutor));
+        _grant(REGISTER_ROLE, address(_initialExecutor));
+        _registerStandard(ERC3000_EXEC_INTERFACE_ID);
     }
 
     receive () external payable {
         emit ETHDeposited(msg.sender, msg.value);
     }
 
+    fallback () external {
+        _handleCallback(msg.sig, msg.data);
+    }
```

Comment:
I'm not sure I fully understand the propose of `handleCallback` as the fallback method for `Govern`.

Reply:
We are basically allowing to install new `onXXX` (e.g. ERC721, 1155) callbacks into Govern at runtime. We catch certain function sigs in the callback, log the data it has and then return a magic number

Reply:
Wow this is done at runtime! Maybe I'm confusing the subjects here but the logic for handling those new standards is happening off chain and we only check the magic number is correct? If that's the case this new paradigm of stateless Ethereum is really impressive.